Improve Keys/ID

DonRichards opened the pull request:
**GitHub Issue**: N/A

# What does this Pull Request do?
Adds __ID__ to the CSV and changes the `keys` to point to __ID__ instead of __file__

# What's new?
- CSV column __ID__
- migration_lookup source to __ID__

# How should this be tested?
Run the migration and verify it migrated.

# Additional Notes:
N/A

# Interested parties
@Islandora/8-x-committers

DonRichards committed:
Update migration.csv

rosiel commented:
comments from Tech Call:
- maybe next step would be to rename 'file' column to 'file_name' or 'file_path'? 
- and maybe add a line in the CSV that is a collection (no file)? might be cleaner to make a separate migration for collections.
- also don't forget to change the documentation!

DonRichards commented:
Pending: Changing `file` to `filename`. Adding __collections__ migrations and __subjects__ migrations.

DonRichards committed:
Add by ID
Update migrate_plus.migration.example_migration_media.yml
Update migrate_plus.migration.example_migration_node.yml
Update migrate_plus.migration.example_migration_file.yml
Update migration.csv
Change file to filename & resolves naming conflict

DonRichards committed:
Adding 2 lines
Create collections.csv
Create migrate_plus.migration.collections_file.yml
Create migrate_plus.migration.collections_media.yml
Rename migrate_plus.migration.collections_file.yml to migrate_plus.migration.example_collections_file.yml
Update and rename migrate_plus.migration.collections_media.yml to migrate_plus.migration.example_collections_media.yml
Create migrate_plus.migration.example_collections_node.yml
Create migrate_plus.migration.collections_subjects.yml
Create collection_subjects.csv
Update and rename migrate_plus.migration.collections_subjects.yml to migrate_plus.migration.example_collections_subjects.yml
Update collections.csv
Update migrate_plus.migration.example_collections_node.yml
Update migrate_plus.migration.example_collections_file.yml
Update migrate_plus.migration.example_collections_file.yml
Update migrate_plus.migration.example_collections_node.yml
Update migrate_plus.migration.example_collections_media.yml
Update migrate_plus.migration.example_collections_file.yml
Update migrate_plus.migration.example_collections_node.yml
Includes requested features to show

DonRichards committed:
lots added
